resources/ext.wikibase.facetedsearch.js

```diff
@@ -140,19 +157,33 @@ function updateErrorState( input ) {
  *
  * @param {HTMLDivElement} facet
  * @param {string} propertyId
+ * @param {string} mode
  * @return {string[]}
  */
-function getListFacetQuerySegments( facet, propertyId ) {
-	const segments = [];
+function getListFacetQuerySegments( facet, propertyId, mode ) {
+	const checkedValues = [];
+
 	[ ...facet.querySelectorAll( '.wikibase-faceted-search__facet-item' ) ].forEach( ( facetItem ) => {
 		const checkbox = facetItem.querySelector( '.cdx-checkbox__input' );
 		if ( !checkbox || !checkbox.checked || !checkbox.value ) {
 			return;
 		}
-		// TODO: Support other operators
-		// TODO: Support OR values
-		segments.push( `haswbfacet:${ propertyId }=${ checkbox.value }` );
+		checkedValues.push( checkbox.value );
 	} );
+
+	if ( checkedValues.length === 0 ) {
+		return [];
+	}
+
+	const segments = [];
+	if ( mode === 'AND' ) {
+		checkedValues.forEach( ( value ) => {
+			segments.push( `haswbfacet:${ propertyId }=${ value }` );
+		} );
+	} else {
+		segments.push( `haswbfacet:${ propertyId }=${ checkedValues.join( '|' ) }` );
+	}
+
```

Comment:
This function seems to be doing two different things: finding selected values in the DOM, and building query segments.

Reply:
Yes that's right. I can refactor it into two functions in a follow up patch

Reply:
Sounds good